File: bbot_controller/scripts/bbot_controller_node.py
#!/usr/bin/env python3
import rospy
from bbot_controller.bbot_controller import Bbot_controller

if __name__ == '__main__':
    rospy.init_node('bbot_controller')
    wheel_radius = rospy.get_param("~wheel_radius")
    wheel_separation = rospy.get_param("~wheel_separation")
    bbot_controller = Bbot_controller(wheel_radius,wheel_separation)

    rospy.spin()

File: bbot_controller/src/bbot_controller_node.cpp
#include <ros/ros.h>
#include "bbot_controller/bbot_controller.h"


int main(int argc, char **argv)
{
    ros::init(argc, argv, "bbot_controller");
    ros::NodeHandle nh;
    ros::NodeHandle pnh("~");
    double wheel_radius, wheel_separation;
    pnh.getParam("wheel_radius", wheel_radius);
    pnh.getParam("wheel_separation", wheel_separation);
    Bbot_controller controller(nh, wheel_radius, wheel_separation);
    ros::spin();

    return 0;
}